### PR TITLE
Love: Added <hui-image> element for handling different image sources

### PR DESCRIPTION
**Pull request by jeradM**
Added camera_image config option that accepts a camera entity.

### CODE REVIEW

**Comment by c727** [comment]
The suggestion was to create a new custom element, like `<hui-image hass=[[hass]] config=[[config]]>`, that is reusable in other cards

**Comment by c727** [comment]
Here we already have a image logic based on state that you can expand to use update Interval if entity domain = camera.

Next step is better support for media_player cover images

https://github.com/home-assistant/home-assistant-polymer/blob/master/src/panels/lovelace/cards/hui-picture-entity-card.js#L95

**Comment by jeradM** [comment]
Oh I thought we were talking about a separate PR for hui-image. I can go ahead on see if I can get that done today and add it to this PR

**Comment by balloob** [comment]
I think that it does not make a lot of sense to add support like this and then create a generic solution. Let's go straight for hui-image.

File: src/panels/lovelace/components/hui-image.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    return {
+      hass: Object,
+      state: {
+        type: String,
+        observer: '_stateChanged'
+      },
+      image: String,
+      stateImage: Object,
+      cameraImage: String,
+      imageSrc: String,
```
[the remark this answers is not in the document]
Comment (jeradM) [reply src/panels/lovelace/components/hui-image.js]: Image is the `image` value from config. imageSrc is the src binding to the `img` element - could be `image`, `state_image[state]`, or the current camera image. Should I do this differently?

File: src/panels/lovelace/common/is-object.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
+// Check if given obj is a JS object and optionally contains all required keys
+export default function isObject(obj, requiredKeys = []) {
```
Comment (balloob) [inline src/panels/lovelace/common/is-object.js]: Let's not call this `isObject` because it does more

Comment (jeradM) [reply src/panels/lovelace/common/is-object.js]: `isValidObject`?  :man_shrugging:

**Pull request merged**